### PR TITLE
Interface rework

### PR DESCRIPTION
**Pull request by lutches**
removed ICombatCreature which was a bad interface, is now replaced with a superclass and a few smaller interfaces that better follow the interface segregation principle

### CODE REVIEW

File: core/src/main/java/com/g16/feyrune/model/combat/AttackHandler.java [hunk 1]
```diff
@@ -14,9 +14,9 @@ public class AttackHandler {
      * @param defender The monster trying to evade the attack
      * @param attack   The specific attack being used
      */
-    public static void handleAttack(ICombatCreature attacker, ICombatCreature defender, IAbility attack) {
+    public static void handleAttack(CombatCreature attacker, CombatCreature defender, IAbilitable attack) {
         if (evasiveManoeuvre(defender, attack)) {
-            int damage = calculateDamage(attacker, defender, attack);
+            int damage = attacker.attack(attack);
```
Comment (Krysset) [inline core/src/main/java/com/g16/feyrune/model/combat/AttackHandler.java:19]: 🤔

Comment (x183) [reply core/src/main/java/com/g16/feyrune/model/combat/AttackHandler.java:19]: yes

File: core/src/main/java/com/g16/feyrune/view/combat/CombatGraphicsRenderer.java [hunk 2]
```diff
@@ -5,14 +5,15 @@
 import com.badlogic.gdx.math.Vector2;
 import com.g16.feyrune.interfaces.ICreature;
 import com.g16.feyrune.model.combat.creatures.EnemyCreature;
+import com.g16.feyrune.model.creature.BaseCreature;
```
Comment (Krysset) [inline core/src/main/java/com/g16/feyrune/view/combat/CombatGraphicsRenderer.java:8]: unused import?

File: core/src/main/java/com/g16/feyrune/view/combat/CombatGraphicsRenderer.java [hunk 3]
```diff
@@ -5,14 +5,15 @@
...
 import com.g16.feyrune.model.player.Player;
 import com.g16.feyrune.model.overworld.encounter.Encounter;
 
 import java.util.ArrayList;
 
 public class CombatGraphicsRenderer {
     private Player player;
-    private ICreature enemyCreature;
+    private EnemyCreature enemyCreature;
```
Comment (Krysset) [inline core/src/main/java/com/g16/feyrune/view/combat/CombatGraphicsRenderer.java:16]: Should probably be CombatCreature

File: core/src/main/java/com/g16/feyrune/view/combat/CombatGraphicsRenderer.java [hunk 4]
```diff
@@ -23,14 +24,14 @@ public class CombatGraphicsRenderer {
 
 
 
-    public CombatGraphicsRenderer(ICreature enemyCreature, Player player){
+    public CombatGraphicsRenderer(EnemyCreature enemyCreature, Player player){
```
Comment (Krysset) [inline core/src/main/java/com/g16/feyrune/view/combat/CombatGraphicsRenderer.java:27]: Should probably also be CombatCreature

File: core/src/main/java/com/g16/feyrune/model/player/Player.java [hunk 5]
```diff
@@ -53,8 +54,8 @@ public Texture getTexture() {
     }
 
     //TODO: This is wrong and temp
-    public ICreature getCreature(){
-        return creature;
+    public PlayerCreature getCreature(){
+        return new PlayerCreature((BaseCreature) this.creature);
```
Comment (Krysset) [inline core/src/main/java/com/g16/feyrune/model/player/Player.java:58]: Confusing function name... Either rename to getPlayerCreature or change to return a creature

File: core/src/main/java/com/g16/feyrune/model/combat/CombatModel.java [hunk 6]
```diff
@@ -13,9 +14,9 @@
 
 public class CombatModel {
     private final Player player;
-    private final ArrayList<ICombatCreature> combatCreatures;
+    private final ArrayList<CombatCreature> combatCreatures;
     private ArrayList<Integer> savedCombatCreatureSpeed;
-    private ArrayList<ICombatCreature> turnOrder;
+    private ArrayList<CombatCreature> turnOrder;
```
Comment (Supergamer1337) [inline core/src/main/java/com/g16/feyrune/model/combat/CombatModel.java:19]: Why arraylist????

File: core/src/main/java/com/g16/feyrune/model/player/Player.java (hunk 5 above)
Comment (Supergamer1337) [reply core/src/main/java/com/g16/feyrune/model/player/Player.java:58]: Agreed

File: core/src/main/java/com/g16/feyrune/model/combat/CombatModel.java (hunk 6 above)
Comment (x183) [reply core/src/main/java/com/g16/feyrune/model/combat/CombatModel.java:19]: Because we push and pop to the list

Comment (Supergamer1337) [reply core/src/main/java/com/g16/feyrune/model/combat/CombatModel.java]: But then it should be List

**Review by Supergamer1337 — APPROVED** [review]
Looks good

**Pull request merged**